components/admin_console/permission_schemes_settings/guest_permissions_tree/guest_permissions_tree.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import Permissions from 'mattermost-redux/constants/permissions';
+import {Role} from 'mattermost-redux/types/roles';
+import {ClientLicense} from 'mattermost-redux/types/config';
+
+import PermissionGroup from '../permission_group.jsx';
+import EditPostTimeLimitButton from '../edit_post_time_limit_button';
+import EditPostTimeLimitModal from '../edit_post_time_limit_modal';
+
+type Props = {
+    license: ClientLicense;
+    onToggle: (role: string, permissions: string[]) => void;
+    readOnly: boolean;
+    scope: string;
+    selectRow: (permission: string) => void;
+    parentRole?: Role;
+    selected?: string | null;
+    role?: Partial<Role>;
+}
+
+const GuestPermissionsTree = ({license, onToggle, readOnly, scope, selectRow, parentRole, selected, role = {permissions: []}}: Props) => {
+    const [editTimeLimitModalIsVisible, setEditTimeLimitModalIsVisible] = React.useState(false);
+
+    const openPostTimeLimitModal = useCallback(() => {
+        setEditTimeLimitModalIsVisible(true);
+    }, []);
+    const closePostTimeLimitModal = useCallback(() => {
+        setEditTimeLimitModalIsVisible(false);
+    }, []);
+    const toggleGroup = useCallback((ids: string[]) => {
+        if (readOnly) {
+            return;
+        }
+        onToggle(role.name!, ids);
+    }, [onToggle, readOnly, role.name]);
+
+    const ADDITIONAL_VALUES = useMemo(() => {
+        return {
+            guest_edit_post: {
+                editTimeLimitButton: (
+                    <EditPostTimeLimitButton
+                        onClick={openPostTimeLimitModal}
+                        isDisabled={readOnly}
+                    />
+                ),
+            },
+        };
+    }, [openPostTimeLimitModal, readOnly]);
+
+    let permissions = useMemo(() => {
+        return [
+            Permissions.CREATE_PRIVATE_CHANNEL,
+            Permissions.EDIT_POST,
+            Permissions.DELETE_POST,
+            Permissions.CREATE_POST,
```

Comment:
This was outside of initial variable declaration but as it was always added after it, I moved it to initial decleration